packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -295,25 +295,40 @@ function mountCustomElement(
     }
 }
 
-function patchCustomElement(n1: VCustomElement, n2: VCustomElement, renderer: RendererAPI) {
-    const elm = (n2.elm = n1.elm!);
-    const vm = (n2.vm = n1.vm);
+function patchCustomElement(
+    n1: VCustomElement,
+    n2: VCustomElement,
+    parent: ParentNode,
+    renderer: RendererAPI
+) {
+    if (n1.ctor !== n2.ctor) {
+        // If the constructor, unmount the current component and mount a new one using the new
+        // constructor.
```

Comment:
```suggestion
        // If the constructor is different, unmount the current component and mount a new one 
        // using the new constructor. This can happen with e.g. lwc:dynamic nodes.
```